microsoft-edge/webview2/how-to/prerelease-testing.md

```diff
@@ -0,0 +1,258 @@
+---
+title: Prerelease testing using preview channels
+description: How to test your WebView2 app by using the preview channels of Microsoft Edge (Edge Canary, Edge Dev, or Edge Beta) to find issues that will affect your particular WebView2 app, before WebView2 changes migrate to the Evergreen WebView2 Runtime.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.service: microsoft-edge
+ms.subservice: webview
+ms.date: 02/15/2024
+---
+# Prerelease testing using preview channels
+
+_Prerelease testing_ means testing your WebView2 app by using the preview channels of Microsoft Edge to find issues that will affect your particular WebView2 app, before WebView2 changes migrate to Evergreen WebView2 Runtime.  Prerelease testing helps your app catch bugs in prerelease channels of WebView2, before the WebView2 changes ship in a stable release and degrade the app experience for end users.  In prerelease testing, you validate your app against early builds of WebView2, to ensure that your app will work with the upcoming changes.
+
+To validate your app against early builds of WebView2, do automated testing with the preview channels of Microsoft Edge.  A preview channel contains a WebView2 preview runtime.
+
+The preview (or "insider") channels of Microsoft Edge consist of:
+* Edge Canary - releases daily.
+* Edge Dev - releases weekly.
+* Edge Beta - releases every 4 weeks.
+
+See also: 
+* [Overview of the Microsoft Edge channels](/deployedge/microsoft-edge-channels)
+* [Become a Microsoft Edge Insider](https://www.microsoft.com/edge/download/insider) - download Edge Canary, Edge Dev, or Edge Beta
+* [Self-hosting by deploying preview channels](./self-hosting.md)
+
+
+<!-- ====================================================================== -->
+## Importance of prerelease testing
+
+Evergreen WebView2 is based on the evergreen Chromium platform, which receives monthly major updates.  To validate your own, particular app against incoming browser updates, test your WebView2 app against the prerelease channels of Microsoft Edge (Edge Canary, Edge Dev, or Edge Beta).  Regressions in WebView2 apps tend to be application-specific, rather than affecting all WebView2 apps.
+
+The only way to catch app-specific regressions before WebView2 changes ship in Edge Stable is to test your particular app against the prerelease channels of Microsoft Edge.  WebView2 is tested in a variety of general scenarios, but it's not feasible for that general testing to cover all the scenarios that various specific apps might encounter.
+
+Many issues in specific WebView2 apps can be discovered before WebView2 changes reach the Stable WebView2 Runtime, by doing prerelease testing, such as in Edge Canary.  For example, the following issues have been caught:
+- [Unable to open https://www.facebook.com, once load will directly ProcessFailed](https://github.com/MicrosoftEdge/WebView2Feedback/issues/4281)
```

Comment:
`For example, the following issues have been caught: `-  but you show the issues which are caught in stable. We probably want good examples which caught in prerelease and then - issues which were not caught and caused major outage for the apps.

Reply:
@vbryh-msft I think there was some confusion between me and @mikehoffms here.

I think what should be stated here is:
> Some examples of the issues that could have been caught in a preview runtime prior to stable include:

So the examples are correct. I'm trying to point out that these issues went into stable, but could have been caught if people did prerelease testing

Reply:
@Reezaali - i'm concerned that listing & linking to these specific mistakes that people made, is too-specific detail & too negative, to include in the main docs.

Reply:
Discussed offline.